source/frontend/src/features/mapSideBar/lease/tabs/consultations/edit/ConsultationAddContainer.tsx

```diff
@@ -0,0 +1,69 @@
+import axios, { AxiosError } from 'axios';
+import { FormikHelpers } from 'formik';
+import { useHistory, useLocation } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import { useConsultationProvider } from '@/hooks/repositories/useConsultationProvider';
+import { IApiError } from '@/interfaces/IApiError';
+
+import { LeasePageNames } from '../../../LeaseContainer';
+import { IConsultationEditFormProps } from './ConsultationEditForm';
+import { ConsultationFormModel } from './models';
+
+export interface IConsultationAddProps {
+  leaseId: number;
+  onSuccess: () => void;
+  View: React.FunctionComponent<React.PropsWithChildren<IConsultationEditFormProps>>;
+}
+
+const ConsultationAddContainer: React.FunctionComponent<
+  React.PropsWithChildren<IConsultationAddProps>
+> = ({ onSuccess, View, leaseId }) => {
+  const history = useHistory();
+  const location = useLocation();
+
+  const backUrl = location.pathname.split(`/${LeasePageNames.CONSULTATIONS}/add`)[0];
+
+  const {
+    addLeaseConsultation: { execute: addConsultation, loading: addConsultationLoading },
+  } = useConsultationProvider();
+
+  const onCreateError = (e: AxiosError<IApiError>) => {
+    if (e?.response?.status === 400) {
+      toast.error(e?.response.data.error);
+    } else {
+      toast.error('Unable to save. Please try again.');
+    }
+  };
+
+  const handleSubmit = async (
+    values: ConsultationFormModel,
+    formikHelpers: FormikHelpers<ConsultationFormModel>,
+  ) => {
+    try {
+      const consultationSaved = await addConsultation(leaseId, values.toApi());
+      if (consultationSaved) {
```

Comment:
hmm, the only issue I see is that if somehow this is false, no error is presented but also the onSuccess and navigation isn't fired.

Reply:
That is true, however, that's our current pattern in other screens.